src/invoke_training/lora/injection/lora_layer_collection.py

```diff
@@ -0,0 +1,42 @@
+import typing
+
+import torch
+
+from invoke_training.lora.layers import BaseLoRALayer
+
+
+class LoRALayerCollection(torch.nn.Module):
+    """A collection of LoRA layers (with names). Typically used to perform operations on a group of LoRA layers during
+    training.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # A torch.nn.ModuleDict may seem like a more natural choice here, but it does not allow keys that contain '.'
```

Comment:
Would a normal `dict[str, torch.nn.ModuleList]` work? I just think depending on the order of two arrays matching could lead to some weird edge cases.

Reply:
Unfortunately not, module in a dict would not be properly registered with the parent torch.nn.Module, so methods like `.train()`, `.parameters()`, etc. wouldn't work. I've updated the comment to make this more clear in [1cee5a7](https://github.com/invoke-ai/invoke-training/pull/5/commits/1cee5a7cdbd036576943808b7c028b50f4c3c288)